File: backend/src/main/java/me/iamhardyha/bugbuddy/auth/GitHubOAuth2UserInfo.java
package me.iamhardyha.bugbuddy.auth;

import java.util.Map;

public class GitHubOAuth2UserInfo {

    private final Map<String, Object> attributes;

    public GitHubOAuth2UserInfo(Map<String, Object> attributes) {
        this.attributes = attributes;
    }

    public String getSubject() {
        Object id = attributes.get("id");
        if (id == null) throw new IllegalStateException("GitHub OAuth2: id missing");
        return id.toString();
    }

    public String getEmail() {
        Object email = attributes.get("email");
        return email instanceof String ? (String) email : null;
    }

    public String getLogin() {
        Object login = attributes.get("login");
        if (!(login instanceof String)) throw new IllegalStateException("GitHub OAuth2: login missing");
        return (String) login;
    }
}

File: backend/src/main/kotlin/me/iamhardyha/bugbuddy/auth/GitHubOAuth2UserInfo.kt
package me.iamhardyha.bugbuddy.auth

class GitHubOAuth2UserInfo(private val attributes: Map<String, Any>) {
    val subject: String
        get() = attributes["id"]?.toString() ?: error("GitHub OAuth2: id missing")

    val email: String?
        get() = attributes["email"] as? String

    val login: String
        get() = attributes["login"] as? String ?: error("GitHub OAuth2: login missing")
}
